tekdoc/main.go

```diff
@@ -17,15 +17,67 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
+	"flag"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme"
+	"io/ioutil"
+	"log"
+	"os"
+	"text/template"
+)
 
-	_ "github.com/tektoncd/pipeline/pkg/pod"
+var (
+	filename = flag.String("f", "", "Name of the file to parse")
 )
 
+type a struct {
+	Taskname  string
+	Input     []v1alpha1.ParamSpec
+	Resources []v1alpha1.TaskResource
+	Output    []v1alpha1.TaskResource
+}
+
 func main() {
-	fmt.Println("Hello this is STILL TekDoc")
 
 	// TODO: take a file path, parse it as YAML -> v1alpha1.Task
+	flag.Parse()
+	dat, err := ioutil.ReadFile(*filename)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Let's add some context in the error logs throughout this file. e.g.

```go
log.Fatalf("error reading file: %v", err)
```

```go
log.Fatalf("error decoding Task: %v", err)
```

etc.